### ISSUE TITLE
Game crashed when I tried to save a new waypoint

### ISSUE THREAD

**Issue by RottMo** [anegostudios/VintageStory-Issues#477]
**Game Version:** 1.12.9-rc.1
**Platform:** Windows
**Modded:** No

### Description
When I tried to save a waypoint on the map, the game crashed.

I tried to log on to the server again, but it crashed while loading.

After restarting the server, I came back into the game, but when I tried to save a waypoint again, it crashed again.

### How to reproduce
1. Open the card (M)
2. Create a new waypoint (right mouse button)
3. Choose a color (for me it was "gold")
4. Give a name (I called it "Baumaterialien")
5. Choose an Icon (I used the Trader Icon)
6. Click on Save
7. Game crashes

### Expected behavior
So far it has always created a new waypoint.

### Logs
```
Version: v1.12.9-rc.1 (Unstable)17.03.2020 23:01:41: Critical error occurred
System.NullReferenceException: Der Objektverweis wurde nicht auf eine Objektinstanz festgelegt.
   bei Vintagestory.GameContent.GuiElementMap.TranslateWorldPosToViewPos(Vec3d worldPos, Vec2f& viewPos)
   bei Vintagestory.GameContent.WaypointMapComponent.Render(GuiElementMap map, Single dt)
   bei Vintagestory.GameContent.GuiElementMap.RenderInteractiveElements(Single deltaTime)
   bei Vintagestory.API.Client.GuiComposer.Render(Single deltaTime)
   bei Vintagestory.API.Client.GuiDialog.OnRenderGUI(Single deltaTime)
   bei Vintagestory.GameContent.GuiDialogWorldMap.OnRenderGUI(Single deltaTime)
   bei _acjXn6niaFrFTsWp4lBHpxwWs8e._JJhGuzS9mpFxXpN7ACoTwC46EoG(Single )
   bei _rhQWE6d6iastcrJCse7YksVlPWM._gPrQZxIytxCnTSan469lVXfaXIb(EnumRenderStage , Single )
   bei _0PCWapRef1vEOuo9wrzYjbApsWP._gPrQZxIytxCnTSan469lVXfaXIb(EnumRenderStage , Single )
   bei _0PCWapRef1vEOuo9wrzYjbApsWP._HqQ6xmbonrWbKBv4qxHBgg1rKPg(Single )
   bei _fnZmbcAsTCacdwTnohz42mF4Rru._HqQ6xmbonrWbKBv4qxHBgg1rKPg(Single )
   bei _hYYecmXIfeEv1EeeGI8DKI5gWw0._Bqh1nPqSWjqLIyzBEo82iV1beTT(Single )
   bei _hYYecmXIfeEv1EeeGI8DKI5gWw0._ZVmpAWXreUTLcpRX8BmkgwPvWYE(Single )
   bei Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(Object sender, FrameEventArgs e)
   bei System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   bei OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp)
   bei OpenTK.GameWindow.DispatchRenderFrame()
   bei OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second)
   bei _aSKeZCetYYAm51Q4VZot6Gi56id._a2RYtGL0S4I7E8yf9neiItsZMQe(_Enwv1jkElPphVcS8xfCUAIYX8ji , String[] )
   bei _4qMy380MiLkPXEpdMSXiTrBhrQp._a2RYtGL0S4I7E8yf9neiItsZMQe(ThreadStart )
```

**Comment by tyronx** [comment]
hm, cannot reproduce this crash. Did you connect to a v1.12.8 server? That might be the cause

**Comment by RottMo** [comment]
No. The server is on the same version. My friend started it and he always updates it to the latest version as soon as there is one.

Incidentally, the error also occurred after reinstalling the game.
However, I have to admit that the error does not really bother me and as long as nobody else has this problem, the error apparently lies either with me or on the server.

I'll just ask my friend about it. :)

Thank you for the time and effort. :D

Best regards
Monika

Von: Tyron Madlener <notifications@github.com>
Gesendet: Mittwoch, 18. März 2020 11:05
An: anegostudios/VintageStory-Issues <VintageStory-Issues@noreply.github.com>
Cc: RottMo <MRottmann@outlook.at>; Author <author@noreply.github.com>
Betreff: Re: [anegostudios/VintageStory-Issues] Game crashed when I tried to save a new waypoint (#477)


hm, cannot reproduce this crash. Did you connect to a v1.12.8 server? That might be the cause

**Comment by Bloodsaw** [comment]
Hmm I tried to reproduce this, only thing I found that in 1.12.9-rc1 the decimal separator issue is now the other way round (not working with , as decimal separator). All of my waypoints had wrong values outside the boundaries of my world size (1mil x 1mil and values with -2000000000)
Maybe the crash (TranslateWorldPosToViewPos(Vec3d worldPos, Vec2f& viewPos)) is connected with overflowing values duo too the another bug?  #455 Just an idea.

**Comment by tyronx** [comment]
It might be related, but I wouldn't know how or why. 

Yes, I'm now enforcing the use of '.' as a decimal separator everywhere. This is less of a headache to handle than culture specific decimal seperators

**Comment by tyronx** [comment]
ooooooh, this crash would happen if a way point position is null, which would indeed happen when you connect to a v1.12.8 server with a v1.12.9 client, due to the fact that the /waypoint add command syntax changed. 
In other worlds this will be fixed in v1.12.9. That version now requires the server to be on v1.12.9
